### ISSUE TITLE
Issue between TurnContext from botbuilder-core & botbuilder-ai

### ISSUE THREAD

**Issue by simon-tannai** [microsoft/botbuilder-js#2130]
Hello ! 

I have a TS error about `TurnContext` given into QnA Maker `getAnswers()` method:
```javascript
this.qnaMaker.getAnswers(stepContext.context),
```

I get this error:
```
Argument of type 'import("/node_modules/botbuilder-core/lib/turnContext").TurnContext' 
is not assignable to parameter of type 
'import("/node_modules/botbuilder-ai/node_modules/botbuilder-core/lib/turnContext").TurnContext'.
 Property 'bufferedReplyActivities' is missing in type 
'import("/node_modules/botbuilder-core/lib/turnContext").TurnContext' 
but required in type 
'import("/node_modules/botbuilder-ai/node_modules/botbuilder-core/lib/turnContext").TurnContext'.ts(2345)
```

It's seems `TurnContext` from `botbuilder-core` and `botbuilder-ai` are not the same.
And I cannot import `TurnContext` from `botbuilder-ai` because it is not exposed:
```javascript
import { LuisRecognizer, QnAMaker, TurnContext } from 'botbuilder-ai';
````
Give me this error: 
`/node_modules/botbuilder-ai/lib"' has no exported member 'TurnContext'`

The only way to fix it I found is to type as `any` the `stepContext.context`:
```javascript
this.qnaMaker.getAnswers(<any>stepContext.context)
```

Versions of packages from my `package.json`:
```json
"botbuilder": "~4.7.0",
"botbuilder-ai": "^4.7.0",
"botbuilder-dialogs": "~4.7.0",
```

Is there a better way to fix it ?

**Comment by Zerryth** [comment]
TL;DR upgrade your Bot Framework dependencies to "~4.8.0" in your package.json
___
Looks like there was a change in the member's name from `bufferedReplies` to `bufferedReplyActivities`.

Anyways, there's a difference between the 2 TurnContexts pulled into `botbuilder` and `botbuilder-ai` because of this, since you have `~` vs. ~`^`, between the two packages (both have a a dependency on `botbuilder-core: 4.1.6`) -- I'll talk to team regarding this if we need to do anything with the dependencies.
___
Anyways, was able to get passed the issue just testing out on the `13.core-bot` sample and adding `QnAMaker` as well.

*package.json* (I didn't edit this--left values as they are currently in sample)
![image](https://user-images.githubusercontent.com/35248895/80619595-c13bb880-89f9-11ea-8b10-7515b5b24f1f.png)

*dialogBot.ts*
```ts
import { QnAMaker } from 'botbuilder-ai';

...

        this.onMessage(async (context, next) => {
            if (context.activity.text === "yo") {
                this.qnaMaker = new QnAMaker({
                    knowledgeBaseId: process.env.QnAKnowledgebaseId,
                    endpointKey: process.env.QnAEndpointKey,
                    host: process.env.QnAEndpointHostName
                });

                const qnaResults = await this.qnaMaker.getAnswers(context);

                // If an answer was received from QnA Maker, send the answer back to the user.
                if (qnaResults[0]) {
                    await context.sendActivity(qnaResults[0].answer);
    
                // If no answers were returned from QnA Maker, reply with help.
                } else {
                    await context.sendActivity('No QnA Maker answers were found.');
                }
            } else {
                console.log('Running dialog with Message Activity.');
    
                // Run the Dialog with the new message Activity.
                await (this.dialog as MainDialog).run(context, this.dialogState);
            }
```
Note that I just currently have it so that if the user sends "`yo`" to the bot, it'll trigger a call to QnA Maker, for testing purposes.

*Emulator*
![image](https://user-images.githubusercontent.com/35248895/80620018-5179fd80-89fa-11ea-836c-5f49a26b9726.png)
Both the regular dialog with LUIS as well as my addition of QnA Maker work.
